### PR TITLE
store: simplify api url config

### PR DESCRIPTION
**Pull request by atomatt**

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -956,11 +922,10 @@ type SnapSpec struct {
...
-	// get the query before doing Parse, as that overwrites it
-	query := s.detailsURI.Query()
-	u, err := s.detailsURI.Parse(snapSpec.Name)
-	if err != nil {
-		return nil, err
+	query := url.Values{}
+
+	if len(s.detailFields) != 0 {
+		query.Set("fields", encodeFields(s.detailFields))
 	}
```
Comment (pedronis) [inline store/store.go]: wondering if instead of the global helper encodeFields, this bit shouldn't be some kind of helper on Store itself

**Comment by mvo5** [comment]
Spellchecker fails with: `store/store.go:169:7: "overriden" is a misspelling of "overridden"`

File: store/store.go (hunk 1 above)
Comment (atomatt) [reply store/store.go]: I extracted this setup to Store.defaultSnapQuery, thanks.

**Comment by chipaca** [comment]
(thank you)

**Pull request merged**